### PR TITLE
test(gui-test): ensure sync does not create duplicate Unicode folders

### LINKED ISSUE #609: QA: Integration test for unicode normalisation
Implement a squish test that validates the behaviour of https://github.com/opencloud-eu/desktop/issues/606
During the development of the test, a v3.0.0-beta.3 build can be used to recreate the undesired behaviour.

Create a folder on the server called  `Öü` do the same locally, start a sync.
The result should be a single folder on the server and not two with different encodings.

### PR DESCRIPTION
**Pull request by Asmitapaudel**
Fixes: https://github.com/opencloud-eu/desktop/issues/609

## Summary

Add an integration test for Unicode folder normalization during synchronization.

## Changes

* Added a Gherkin scenario to verify that syncing the same Unicode folder from the server and client does not create duplicates.
* Updated the `get_folder_items_count()` helper function to correctly handle the root folder and exclude the default `Shares` folder.

### CODE REVIEW

**Comment by guruz** [comment]
Would be interesting if running this test on MacOS vs Linux behaves differently... (NFC vs NFD etc)

File: test/gui/features/sync-edge-cases/syncResources.feature [hunk 1]
```diff
@@ -219,3 +219,28 @@ Feature: Syncing files
...
+
+    Scenario: Synchronize Unicode folder names without creating duplicates
+        Given user "Alice" has set up a client with default settings
+        And the user has paused the file sync
+        And user "Alice" has created folder "Öü" in the server
+        And user "Alice" has uploaded file with content "openCloud test" to "Öü/testFile.txt" in the server
+        When user "Alice" creates a folder "Öü" inside the sync folder
+        # And user "Alice" creates a folder "Öü/another" inside the sync folder
+        # Given the user has created a file "Öü/newfile.txt" with size "50MB" in the sync folder
+        When user "Alice" creates a file "Öü/newfile.txt" with the following content inside the sync folder
```
Comment (saw-jan) [inline test/gui/features/sync-edge-cases/syncResources.feature:230]:
```suggestion
        When user "Alice" creates a folder "Öü" inside the sync folder
        And user "Alice" creates a file "Öü/newfile.txt" with the following content inside the sync folder
```

File: test/gui/features/sync-edge-cases/syncResources.feature [hunk 2]
```diff
@@ -219,3 +219,28 @@ Feature: Syncing files
...
+        #     | files           |
+        #     | /Öü/newfile.txt |
+        When the user resumes the file sync on the client
+        # And the user waits for the files to sync
+        # And the user waits for file "Öü/newfile.txt" to be synced
+        And the user force syncs the files
+        Then as "Alice" only one folder named "Öü" should exist on the server
+        And as "Alice" file "Öü/testFile.txt" should exist in the server
+        # And as "Alice" folder "Öü/another" should exist in the server
+        And as "Alice" file "Öü/newfile.txt" should exist in the server
```
Comment (saw-jan) [inline test/gui/features/sync-edge-cases/syncResources.feature]:
```suggestion
        And the user waits for "1" seconds
        And the user resumes the file sync on the client
        And the user waits for the files to sync
        Then the file "Öü/newfile.txt" should exist on the file system
        And as "Alice" file "Öü/newfile.txt" should exist in the server
```

Comment (saw-jan) [inline test/gui/features/sync-edge-cases/syncResources.feature:240]: and lastly check the folder count
```suggestion
        And as user "Alice" folder "/" should contain "1" items in the server
```

File: test/gui/helpers/api/webdav_helper.py [hunk 3]
```diff
@@ -143,3 +144,17 @@ def send_resource_share_invitation(user, resource, sharee, permission_role):
     assert response.status_code in [200, 201], (
         f"Failed to send share invitation: {response.status_code}"
     )
+
+
+def get_matching_folder_count(user, folder_name):
```
Comment (saw-jan) [inline test/gui/helpers/api/webdav_helper.py]: this not needed

File: test/gui/features/sync-edge-cases/syncResources.feature [hunk 4]
```diff
@@ -219,3 +219,28 @@ Feature: Syncing files
         And as "Brian" folder "Shares/simple-folder/sub-folder" should exist in the server
         And as "Brian" file "Shares/simple-folder/simple.pdf" should exist in the server
         And as "Brian" the file "Shares/simple-folder/uploaded-lorem.txt" should have the content "overwrite openCloud test text file" in the server
+
+
+    Scenario: Synchronize Unicode folder names without creating duplicates
```
Comment (saw-jan) [inline test/gui/features/sync-edge-cases/syncResources.feature]:
```suggestion
    Scenario: Sync same unicode folder from server and client at the same time
```

**Comment by saw-jan** [comment]
test is failing:
https://ci.opencloud.rocks/repos/11/pipeline/653/49
```
Failing scenarios:
  features/sync-edge-cases/syncResources.feature:224  Sync same unicode folder from server and client at the same time
```

File: test/gui/helpers/api/webdav_helper.py [hunk 5]
```diff
@@ -56,11 +56,15 @@ def get_folder_items_count(user, folder_name):
     folder_name = folder_name.strip('/')
```
Comment (saw-jan) [inline test/gui/helpers/api/webdav_helper.py:57]:
```suggestion
    folder_name = folder_name.strip('/')
    if folder_name != '':
        folder_name += '/'
```

File: test/gui/helpers/api/webdav_helper.py [hunk 6]
```diff
@@ -56,11 +56,15 @@ def get_folder_items_count(user, folder_name):
...
     root_element = get_folder_items(user, folder_name)
     total_items = 0
+    folder_path = f'{user}/{folder_name}/' if folder_name else f'{user}/'
```
Comment (saw-jan) [inline test/gui/helpers/api/webdav_helper.py]:
```suggestion
```

File: test/gui/helpers/api/webdav_helper.py [hunk 7]
```diff
@@ -56,11 +56,15 @@ def get_folder_items_count(user, folder_name):
...
     for response_element in root_element:
         for href_element in response_element:
             # The first item is folder itself so excluding it
-            if href_element.tag == '{DAV:}href' and not href_element.text.endswith(
-                f'{user}/{folder_name}/'
+            if (
+                href_element.tag == '{DAV:}href'
+                and not href_element.text.endswith(folder_path)
+                and not href_element.text.rstrip('/').endswith('/Shares')
             ):
```
Comment (saw-jan) [inline test/gui/helpers/api/webdav_helper.py]:
```suggestion
            if not href_element.text:
                continue
            is_folder_item = re.search(f'/{user}/{folder_name}.+', href_element.text)
            shares_folder = href_element.text.endswith(f'/{user}/Shares/')
            if href_element.tag == '{DAV:}href' and is_folder_item and not shares_folder:
```

File: test/gui/steps/server_context.py
Comment (saw-jan) [inline test/gui/steps/server_context.py]:
```suggestion
    'as user "{user_name}" folder "{folder_name}" should contain "{items_number}" items in the server'
```

File: test/gui/helpers/api/webdav_helper.py [hunk 8]
```diff
@@ -54,14 +55,19 @@ def get_folder_items(user, resource):
...
 def get_folder_items_count(user, folder_name):
     folder_name = folder_name.strip('/')
+    if folder_name != '':
+        folder_name += '/'
     root_element = get_folder_items(user, folder_name)
     total_items = 0
+
     for response_element in root_element:
         for href_element in response_element:
             # The first item is folder itself so excluding it
```
Comment (saw-jan) [inline test/gui/helpers/api/webdav_helper.py]: the comment is irrelevant now
```suggestion
```

**Review by saw-jan — APPROVED** [review]
